banzai_nres/wavelength.py

```diff
@@ -134,6 +135,23 @@ def refine_wavelengths(self, image):
             for trace_id, ref_id in zip(trace_ids[fiber_ids == fiber], ref_ids[fiber_ids == fiber]):
                 this_trace = np.isclose(image.traces, trace_id)
                 image.wavelengths[this_trace] = wavelength_model(x2d[this_trace], ref_id * np.ones_like(x2d[this_trace]))
+            # Set the true physical order number
+            ref_ids[fiber_ids == fiber] += m0
+        # Calculate which fiber is which.
+        # the first ref_id = 50 (arbitrary, but somewhere about the middle of the detector)
+        # is largest fiber number that is lit.
+        largest_fiber_index = fiber_ids[ref_ids.tolist().index(50)]
+        if image.fiber2_lit:
+            largest_fiber_number = 2
+            smaller_fiber_number = 1
+        else:
+            largest_fiber_number = 1
+            smaller_fiber_number = 0
+
+        true_fibers = [largest_fiber_number if fiber_id == largest_fiber_index else smaller_fiber_number
+                       for fiber_id in fiber_ids]
+        # set fibers attribute on image
+        image.fibers = Table({'trace': trace_ids, 'fiber': true_fibers})
```

Comment:
Just want to verify: should we have fibers be a table or an array? If table, then the use of image.fibers in WeightedExtract (See comment labelled 1 above) may fail. Image.spectrum will have trace_ids and fibers as columns in the extracted spectra if image.fibers is an array.

Reply:
I believe I fixed this.